samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -260,8 +267,9 @@ public void run() {
       MetadataResourceUtil metadataResourceUtil = new MetadataResourceUtil(jobModel, this.metrics, config);
       metadataResourceUtil.createResources();
 
-      // fan out the startpoints if startpoints is enabled
-      if (new JobConfig(config).getStartpointEnabled()) {
+      // fan out the startpoints if startpoints is enabled and if the metadata changed across attempts.
+      // the metadata changed should be false and only get evaluated if job coordinator high availability is enabled.
+      if (new JobConfig(config).getStartpointEnabled() && !metadataChangedAcrossAttempts) {
```

Comment:
what if startpoint is enabled and metadata has changed? we just ignore? as in for AM-HA with metadata changes we dont create startpoint manager? if we are planning another PR for this then lets leave a comment saying tbd.

sorry missed this in the first pass.

Reply:
yeah. We will not fanout startpoint since there may be containers that are already running in which case the newly instantiated containers will act on startpoint but the old ones wont. As a result, we skip fanout and applications will force job restart through new deployment like how they do it today.

Reply:
Thanks for pointing this out. I intended to handle it as part of this PR.
Fixed it and added unit tests.